src/realm/alloc_slab.cpp

```diff
@@ -916,6 +915,10 @@ ref_type SlabAlloc::attach_file(const std::string& path, Config& cfg, util::Writ
     DetachGuard dg(*this);
 
     reset_free_space_tracking();
+
+    // the file could have been produced on a device with a different
+    // page size than our own so don't expect the size to be aligned
+    size = round_up_to_page_size(size);
```

Comment:
This leaves the readers with a view that is larger than what is backed by the file. I think this should be okay, because eventually we will resize the file and call `attach_file()` again. However, I'm not 100% sure about what other side effects this may have between this point and that.

Reply:
Based on the test failures, setting the size to be larger than the file size isn't valid on Windows.

Reply:
Turns out that the windows errors were because I was (incorrectly) rounding up even for non-encrypted files, and this caused our Windows builders to run out of disk space.

I looked into it further and haven't found anything to dispute rounding up here. In other places, the memory mapping is always [rounded up](https://github.com/realm/realm-core/blob/master/src/realm/util/file_mapper.cpp#L567) to a page size regardless of the file size. We should be fine as long as we don't read from that memory which shouldn't happen (if it did, that would indicate a different problem). The other issue I could think of is where the rounded up section becomes a valid block due to a different process extending the file. But that isn't an issue here because we are the initiator process under the interprocess control lock, and we do the file extension ourselves [immediately afterwards](https://github.com/realm/realm-core/blob/master/src/realm/db.cpp#L1312) if necessary.